pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/SSBQueryIntegrationTest.java

```diff
@@ -0,0 +1,265 @@
+
+    // h2 response
+    Assert.assertNotNull(_h2Connection);
+    Statement h2statement = _h2Connection.createStatement(ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY);
+    h2statement.execute(query);
+    ResultSet h2ResultSet = h2statement.getResultSet();
+
+    // compare results.
+    if (h2ResultSet.first()) {
+      for (int i = 0; i < numRows; i++) {
+        for (int c = 0; c < numColumns; c++) {
+          String h2Value = h2ResultSet.getString(c + 1);
+          String pinotValue = resultTableResultSet.getString(i, c);
+          boolean error = ClusterIntegrationTestUtils.fuzzyCompare(h2Value, pinotValue, pinotValue);
+          if (error) {
+            throw new RuntimeException("Value: " + c + " does not match, expected: " + h2Value
+                + ", got pinot value: " + pinotValue);
+          }
+        }
+        if (!h2ResultSet.next()) {
+          return;
+        }
+      }
+    }
+  }
+
+  @Override
+  protected Connection getPinotConnection() {
+    Properties properties = new Properties();
+    properties.put("queryOptions", "useMultistageEngine=true");
+    if (_pinotConnection == null) {
+      _pinotConnection = ConnectionFactory.fromZookeeper(properties, getZkUrl() + "/" + getHelixClusterName());
+    }
+    return _pinotConnection;
+  }
+
+  @Override
+  protected void overrideBrokerConf(PinotConfiguration brokerConf) {
+    brokerConf.setProperty(CommonConstants.Helix.CONFIG_OF_MULTI_STAGE_ENGINE_ENABLED, true);
+    brokerConf.setProperty(QueryConfig.KEY_OF_QUERY_RUNNER_PORT, 8421);
+  }
+
+  @Override
+  protected void overrideServerConf(PinotConfiguration serverConf) {
+    serverConf.setProperty(CommonConstants.Helix.CONFIG_OF_MULTI_STAGE_ENGINE_ENABLED, true);
+    serverConf.setProperty(QueryConfig.KEY_OF_QUERY_SERVER_PORT, 8842);
+    serverConf.setProperty(QueryConfig.KEY_OF_QUERY_RUNNER_PORT, 8422);
+  }
+
+  @AfterClass
+  public void tearDown()
+      throws Exception {
+    // unload all SSB tables to both Pinot and H2.
+
+    // stop components and clean up
+    stopServer();
+    stopBroker();
+    stopController();
+    stopZk();
+
+    FileUtils.deleteDirectory(_tempDir);
+  }
+
+  private void loadSSBToPinot()
+      throws Exception {
+    for (String tableResource : SSB_QUICKSTART_TABLE_RESOURCES.values()) {
+      URL resourceUrl = getClass().getClassLoader().getResource(tableResource);
+      Assert.assertNotNull(resourceUrl);
+      File resourceFile = new File(resourceUrl.getFile());
+      if (!new BootstrapTableTool("http", "localhost", _controllerPort,
+          resourceFile.getAbsolutePath(), null).execute()) {
+        throw new RuntimeException("Failed to bootstrap table " + tableResource);
+      }
+    }
+  }
+
+  @SuppressWarnings({"SqlDialectInspection", "SqlNoDataSourceInspection"})
+  private void loadSSBToH2()
+      throws Exception {
+    setUpH2Connection();
+    Map<String, String> createTableSqlMap = new HashMap<>();
+    createTableSqlMap.put("dates",
```

Comment:
(minor) maybe it makes sense to put these as `.sql` files in the same directory as the pinot spec files? it'd be a bit easier to read.

same comment applies to the data provider, might be nice to be able to just easily verify that it matches the SSB queries by looking at the file.

Reply:
added TODOs